UserDetailsClient/UserDetailsClient.Core/Features/LogOn/B2CAuthenticationService.cs

```diff
@@ -5,34 +5,55 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using Xamarin.Forms;
 
 namespace UserDetailsClient.Core.Features.LogOn
 {
-    public class B2CAuthenticationService : IAuthenticationService
+    /// <summary>
+    ///  For simplicity, we'll have this as a singleton. 
+    /// </summary>
+    public class B2CAuthenticationService
     {
-        public IPublicClientApplication PCA = null;
+        private readonly IPublicClientApplication _pca;
 
-        public object ParentActivityOrWindow { get; set; }
 
-        public B2CAuthenticationService()
+        private static readonly Lazy<B2CAuthenticationService> lazy = new Lazy<B2CAuthenticationService>
+           (() => new B2CAuthenticationService());
+
+        public static B2CAuthenticationService Instance { get { return lazy.Value; } }
+
+
+        private B2CAuthenticationService()
         {
+
             // default redirectURI; each platform specific project will have to override it with its own
-            PCA = PublicClientApplicationBuilder.Create(B2CConstants.ClientID)
+            var builder = PublicClientApplicationBuilder.Create(B2CConstants.ClientID)
                 .WithB2CAuthority(B2CConstants.AuthoritySignInSignUp)
                 .WithIosKeychainSecurityGroup(B2CConstants.IOSKeyChainGroup)
-                .WithRedirectUri($"msal{B2CConstants.ClientID}://auth")
-                .Build();
+                .WithRedirectUri($"msal{B2CConstants.ClientID}://auth");
+
+            // Android implementation is based on https://github.com/jamesmontemagno/CurrentActivityPlugin
+            // iOS implementation would require to expose the current ViewControler - not currently implemented as it is not required
```

Comment:
@jennyf19 - can you review this statement please? Do we want to have an implementation here, for example based on https://forums.xamarin.com/discussion/24689/how-to-acces-the-current-view-uiviewcontroller-from-an-external-service ?